compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -2714,7 +2714,22 @@ impl<'a> Context<'a> {
                     .get_or_create_witness_var(input)
                     .map(|val| self.convert_vars_to_values(vec![val], dfg, result_ids))?)
             }
-            _ => todo!("expected a black box function"),
+            Intrinsic::ArrayToStrLossy => todo!("non-constant array_to_str_lossy"),
```

Comment:
How should this be implemented in acir-gen?

Reply:
Why not use ASCII instead of UTF-8? The translation is much simpler and Noir itself [only supports ASCII](https://github.com/noir-lang/noir/blob/master/compiler/noirc_frontend/src/lexer/lexer.rs#L155).

Reply:
> the Noir language itself [only supports ASCII](https://github.com/noir-lang/noir/blob/master/compiler/noirc_frontend/src/lexer/lexer.rs#L155).

Not for [string literals](https://github.com/noir-lang/noir/blob/28211a397b810c661204b45b7da06f7cad345278/compiler/noirc_frontend/src/lexer/lexer.rs#L412).

Reply:
Ah, I had a PR to only allow ASCII in Noir and it was intended to include string literals.

Reply:
I've been thinking it over some and talking with Maxim and I think moving to only ASCII string values or at least just not checking UTF-8 validity is the way forward. Since we can't really check it in acir without using oracle functions or similar.